ErrorView tmpl_name uses match_extension setting.

AndreasBackx opened the pull request:
ErrorView forced the `.jinja` extension on error template files. It now uses the `match_extension` setting and defaults to `.jinja` if the value evaluates to `False`.

AndreasBackx commented on django_jinja/views/__init__.py:
```diff
@@ -1,11 +1,11 @@
 
 import django
-
-from django.conf import settings
```
I removed this line as it was unused.

niwinz commented on django_jinja/views/__init__.py:
```diff
@@ -46,22 +47,24 @@ def put(self, request, *args, **kwargs):
     def patch(self, request, *args, **kwargs):
         return self.get(request, *args, **kwargs)
 
+    match_extension = get_match_extension() or ".jinja"
```
I think that this should be resolved at runtime and not at class definition time. With your current approach can execute code before django completly initializes...

AndreasBackx replied:
Sorry for the late reply. I can move it to something that initializes via the AppConfig perhaps. Any recommendations?

AndreasBackx commented:
Would it be possible to add an option for the default extension? This is my use case:

I have email templates both in HTML (`.html`) and plaintext (`.txt`) that I both use Jinja2 with. If I set `match_extension` to `.html`. Then it will make the `ErrorView` use `400.html` for example. But then it will not pick up my `.txt` files. So I would want to have `match_extension` to be `None`. And use `.html` for all of my files (including `ErrorView` templates).

The name of the option could be `default_extension` and its default value would be `.jinja` if `match_extension` is `None`. Else it would by default be equal to `match_extension`. Any input?

AndreasBackx commented:
I made a new commit with a simple `default_extension` implementation. Let me hear what you think.

niwinz commented:
Many thanks for you work!

I think I have explained me wrong, the current (latest) approach has the same problem and introduces a new option that is pretty the same as "match_extension", that can lead to confusion to which should be used.

niwinz commented on django_jinja/views/__init__.py:
```diff
@@ -46,22 +47,24 @@ def put(self, request, *args, **kwargs):
     def patch(self, request, *args, **kwargs):
         return self.get(request, *args, **kwargs)
 
+    default_extension = get_default_extension() or ".jinja"
```
With the latest changes, get_default_extension() is again called at compile time, and I think this should be done at runtime.

Don't worry, I'll try to take you code and adapt it.

niwinz commented:
A have opened a #190 pr with your changes adapted to which I have said in the comment.

Please review it, and if it conforms the behavior that you want to match, let me know if everything is ok for merge it.

AndreasBackx commented:
Great, I wanted to avoid having multiple properties like you are using now in #190. So I asked for recommendations but didn't get a reply so I waited for your reply before I changed that.

>I think I have explained me wrong, the current (latest) approach has the same problem and introduces a new option that is pretty the same as "match_extension", that can lead to confusion to which should be used.
You did not explain yourself wrong, I just already implemented what I thought was needed. You could give some feedback on how to change it. This way I could immediately show an example implementation.

However, the problem with #190 is that it while it does allow for you to set the extension on `ErrorView` templates, doing so will remove the ability to use multiple extensions. So having `.txt` and `.html` templates is not possible unless you want to use `.jinja` for `ErrorView` templates. But then we end up with multiple file extensions that are unneeded. I would love to be able to use `.txt` and `.html` and use `.html` for `ErrorView` templates.

niwinz commented:
Hmm, I'm pretty confused right now.

Seems like I'm clearly don't undertand that problem you are trying to solve. Currently without the patch, the error views didn't respected the match_extension option, and this is wrong. Now it is fixed by #190 

Adding an other option for set the default extension seems redundant or at least is how I see it.

We need to think in solve real problems. In my point of view, `.jinja` is a proper extension for template files independently if they generate plain/text files or they generates plain/html files. Because they clearly identifies that that file is jinja template. It just "no or at less possible surprise" principe, just use one exentension for identify template files.

If in your project you want to use multiple extensions for identify just the same thing (jinja templates) I think that you need to fix it on you project, and not in django-jinja. Maybe deactivating the match extension, and using the regular expression matching that allows more flexible way to match files. This approach has one issue right now, is that error views does not allow set custom template names, and this is that I'm going to fix right now.

niwinz commented:
With the https://github.com/niwinz/django-jinja/pull/190/commits/8f2731e4d62598a276bbc401947e6064eeef0692 commit, now you can overwrite the template name:

```python
# yourproject/urls.py
from django_jinja import views

handler400 = views.BadRequest.as_view(tmpl_name="400.html")
# [...]
```

Allowing you to setup proper matching mechanism for your needs and adapt the error views to use that policy overwriting the default template name.

AndreasBackx commented:
@niwinz thanks for the commit, I'll give it a shot when I get my laptop back. I do however prefer using extensions to make clear what the file is used for. The extension `.jinja` does not say anything about the file except that it is a Jinja2 template. Having extensions like `.txt` make it clear that the file is used for plaintext email templates for examples. `.html` is of course for HTML, `.scss` for SCSS files (these can still contain Jinja2 templates) and so forth. Using `.jijna` everywhere makes you lose a lot of context.

niwinz commented:
Yeah, but you can put two extensions, something like `somefile.scss.jinja` :P if the problem is the visual file identification.

AndreasBackx commented:
Then the problem would be that I'd have to change the formatting of every single file in my IDE/text editor. I'd rather have SCSS formatting than Jinja.
